applicator.json

```diff
@@ -0,0 +1,60 @@
+{
+    "$schema": "http://json-schema.org/draft-08/schema#",
+    "$id": "https://json-schema.org/draft-08/applicator",
+    "$recursiveAnchor": true,
+    "$vocabulary": {
+        "https://json-schema.org/draft-08/vocabularies/applicator": true
```

Comment:
I'm confused by this.  Does the URI here imply usage or definition?  If usage, why is the *applicator.json* meta-schema declaring it uses itself?  If definition, why is the value an object (what case would there be to define multiple vocabularies in the same document)?

Reply:
@gregsdennis great questions that get to the heart of why this was such a difficult PR to write up- I expect to have to revise this before it makes sense to people so I'm glad you're focusing on this.

As covered in the intro section with the S, M, M' example, where M is the meta-schema for S, and M' is the meta-schema for M, the vocabularies in `$vocabulary` within M are declaring the expected usage within S.  Meta-schemas describe the expected / allowed behavior of schemas that conform to them.

The whole reason for this is that meta-schemas will often declare usage of multiple vocabularies.  It's extremely rare for only one vocabulary to be sufficient.  The "schema" meta-schema will require at least four, counting core, and the "hyper-schema" meta-schema will require at least five (it depends on what we decide to do with `format` and/or `content*`, which I'll work on after we establish the basics here).

So, most people will never write meta-schemas.  If they do, it will probably be fairly rare.  Many people will write schemas.  Meta-schemas have to deal with the complexity of assembling a usable overall vocabulary from the fine-grained re-usable component vocabularies.  Schema authors do not deal with that- they reference a meta-schema that is documented to do whatever it is they are trying to do.  Most schema authors and users should not care about vocabularies directly, most of the time.

Reply:
@gregsdennis there is a specific point that the core vocabulary need not be explicitly listed.  I'm of two minds on that, maybe we just say that it MUST always be present and it MUST always have a value of `true`.

Reply:
It sounds like `$vocabularies` is kind of an equivalent of the JS `imports` statement for schema keywords.  Unless a keyword is defined within one of the "imported" meta-schemas, a validator will ignore it.

Do you think we're getting close to compiler-like behavior here?

Reply:
It is sort of compiler-like, but let me know if you have any other ideas on how to support machine-detectible re-usuable chunks of functionality.  I've been poking at this idea for most of a year, and no one has proposed a viable alternative of which I am aware (and really I think you're the only person who has given a large amount of feedback - thanks for that!)

I will say that the OpenAPI steering group is very interested in this feature as I have sold them on it as a way to solve their persistent code generation tooling problems.

Reply:
No worries.  I just brought up the compiler thing because we've historically tried to steer away from logic/behavior.

Reply:
@gregsdennis I'm not sure how this qualifies- let me know if I'm missing something that you think we should consider.  There is a decision about whether the implementation supports the vocabularies that the meta-schema needs to have supported, but in many cases `$schema` itself is treated that way already.  Some implementations won't process schemas with unrecognized meta-schemas.

Reply:
@gregsdennis I think I'm leaning towards your position that the core vocabulary MUST be explicitly included and set to true, and that the behavior of anything else is undefined (but it is RECOMMENDED that an error be raised).

It's slightly annoying but arguably no more annoying that anything else in this complex feature, and we can always relax it later easily enough.  Consistency is probably better for now.

Reply:
I've now made it required.